### ISSUE TITLE
Sum on sparse arrays results in bad slicing results

### ISSUE THREAD

**Issue by DieterDePaepe** [scipy/scipy#12879]
There is a strange behavior when `sum`ming sparse arrays along a specific axis. The matrix dimensions and slicing of the result do not work as expected. The behavior contradicts the documentation (which mentions the axis will be dropped) and also does not correspond to behavior as seen in numpy.

#### Reproducing code example:
```python
from scipy.sparse import csr_matrix
csr_matrix((2,3)).sum(axis=0)
# matrix([[0., 0., 0.]])
csr_matrix((2,3)).sum(axis=0)[0]
# matrix([[0., 0., 0.]])
csr_matrix((2,3)).sum(axis=0)[0][0][0][0][0][0]
# matrix([[0., 0., 0.]])
csr_matrix((2,3)).sum(axis=1)
# matrix([[0.],[0.]])
csr_matrix((2,3)).sum(axis=1)[0]
# matrix([[0.]])
csr_matrix((2,3)).sum(axis=1)[0][0][0][0]
# matrix([[0.]])

import numpy as np
np.zeros((2,3)).sum(axis=0)
# array([0., 0., 0.])
np.zeros((2,3)).sum(axis=1)
# array([0., 0.])
```

#### Scipy/Numpy/Python version information:
```
import sys, scipy, numpy
print(scipy.__version__, numpy.__version__, sys.version_info)
1.5.2 1.19.1 sys.version_info(major=3, minor=7, micro=8, releaselevel='final', serial=0)
```

Also tested on scipy 1.5.0 and 1.4.1

**Comment by DieterDePaepe** [comment]
Upon further reading, I noticed scipy works using matrices rather than ndarrays. So the behavior is in fact as expected, as described in [numpy.matrix.sum](https://numpy.org/devdocs/reference/generated/numpy.matrix.sum.html#numpy.matrix.sum).

However, I am not closing this issue, because I still find the [scipy documentation](https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.csr_matrix.sum.html) confusing:

> Returns sum_along_axisnp.matrix
>     A matrix with the same shape as self, with the specified axis removed.

**Comment by perimosocordiae** [comment]
I agree that the docstring can be improved here. The return value description should also mention that a scalar is returned when `axis=None`.

**Comment by kno10** [comment]
I find it irritating to get a matrix back, and usually want an array.
Apparently numpy also by now suggests to prefer using ndarrays instead of matrix: https://numpy.org/doc/stable/reference/generated/numpy.matrix.html
any plans on eventually migrating the sparse matrix classes to produce ndarrays instead of matrixes?

**Comment by jakevdp** [comment]
There are [plans](https://github.com/numpy/numpy/pull/10142) in numpy to deprecate `np.matrix`, so at some sparse matrices will have to produce arrays.

**Comment by perimosocordiae** [comment]
There are efforts underway of varying completeness to support a sparse ndarray-like interface. The one most actively developed right now is https://github.com/pydata/sparse.

We've had some discussion of merging sparse ndarray support into scipy.sparse proper, but for now the plan is to let other packages explore the design space and see what works best.

**Comment by stefanv** [comment]
Desired behavior implemented in #14822. Docstring for `sum` can still be improved.
